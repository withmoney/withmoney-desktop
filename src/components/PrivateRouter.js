import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import checkAuth from 'app/utils/checkAuth';

const PrivateRouter = ({ component: Component, ...otherProps }) => (
  <Route
    {...otherProps}
    render={props => (checkAuth() ? <Component {...props} /> : <Redirect to="/login" />)}
  />
);

PrivateRouter.propTypes = {
  component: PropTypes.any.isRequired,
};

export default PrivateRouter;
