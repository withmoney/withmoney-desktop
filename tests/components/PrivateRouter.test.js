import React from 'react';
import renderer from 'react-test-renderer';
import PrivateRouter from 'components/PrivateRouter';

jest.mock('react-router-dom', () => ({
  Route: 'route',
  Redirect: 'redirect',
}));

describe('PrivateRouter', () => {
  let Component;

  beforeAll(() => {
    Component = global.setup(PrivateRouter);
  });

  it('should render a component correctly', () => {
    const children = <div>children</div>;
    const wrapper = renderer.create(
      Component({
        component: children,
      }),
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
