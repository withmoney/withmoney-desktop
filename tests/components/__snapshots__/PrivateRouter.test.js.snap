// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRouter should render a component correctly 1`] = `
<route
  render={[Function]}
/>
`;
